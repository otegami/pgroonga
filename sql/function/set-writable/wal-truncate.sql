SET pgroonga.enable_wal = true;

CREATE TABLE memos (
  content text
);

INSERT INTO memos VALUES ('Groonga is fast!');

CREATE INDEX pgrn_index ON memos USING PGroonga (content);

SELECT pgroonga_set_writable(false);

SELECT pgroonga_wal_truncate('pgrn_index');

SELECT pgroonga_set_writable(true);

SELECT pgroonga_wal_truncate('pgrn_index');

DROP TABLE memos;

SET pgroonga.enable_wal = default;
