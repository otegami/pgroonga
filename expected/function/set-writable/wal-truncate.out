SET pgroonga.enable_wal = true;
CREATE TABLE memos (
  content text
);
INSERT INTO memos VALUES ('Groonga is fast!');
CREATE INDEX pgrn_index ON memos USING PGroonga (content);
SELECT pgroonga_set_writable(false);
 pgroonga_set_writable 
-----------------------
 t
(1 row)

SELECT pgroonga_wal_truncate('pgrn_index');
 pgroonga_wal_truncate 
-----------------------
                     2
(1 row)

SELECT pgroonga_set_writable(true);
 pgroonga_set_writable 
-----------------------
 f
(1 row)

SELECT pgroonga_wal_truncate('pgrn_index');
 pgroonga_wal_truncate 
-----------------------
                     2
(1 row)

DROP TABLE memos;
SET pgroonga.enable_wal = default;
